aten/src/ATen/native/TensorShape.cpp

```diff
@@ -392,6 +390,86 @@ Tensor cat(TensorList tensors, int64_t dim) {
   return result;
 }
 
+Tensor block_diag(TensorList tensors) {
+  Tensor result;
+  if (tensors.size() == 0) {
+    result = at::empty({});
+    return result;
+  }
+
+  // Use at::result_type to find the right tensor to copy options from, so
+  // that the output data type is correct
+  const Tensor* options_tensor = &tensors[0];
```

Comment:
Sorry, my initial suggestion to change this to `const` was wrong. I realized that you are later reassigning `options_tensor` with `other_tensor` in line 418: `options_tensor = other_tensor;`.

In this case we should `options_tensor` and `other_tensor` both into `Tensor`.

Reply:
This was why I used a const pointer, so that we could keep the Tensor const but also change which tensor we're pointing to. Is it preferable not to enforce that the Tensor stay constant so that we can use the reference syntax?

Reply:
The only downside to using a const pointer here is that it looks strange. Usually when I see a pointer I think "optional argument" or "iterator", none of which are happening here. If you'd like to keep it then go for it